src/lib/app_registry.tsx

```diff
@@ -76,8 +76,12 @@ class Home extends RootContainer {
 class WorksForYou extends RootContainer {
   constructor(props) {
     super(props)
+
+    // This really has to be a boolean for Relay to understand it
+    const showSpecialNotification = props.selectedArtist && props.selectedArtist !== ""
```

Comment:
On second thought, you don’t even really need to set the `showSpecialNotification` flag from here, you can instead use Relay’s `prepareVariables` to derive it from `selectedArtist`: https://facebook.github.io/relay/docs/api-reference-relay-container.html#preparevariables

Something like the following, I think?

```js
  prepareVariables:({ selectedArtist }) => {
    return {
      selectedArtist,
      showSpecialNotification: !!selectedArtist,
    };
  },
```